18 Normalizr/src/pages/student-detail/components/studentForm.tsx

```diff
@@ -24,27 +26,44 @@ export const StudentForm = (props: Props) => {
     props.saveStudent(props.student);
   };
 
+  const onSelectCountry = (event, value) => {
+    let field = event.target.name;
+
+    props.fireFieldValueChanged(props.student.country, field, value);
+  };
 
   return (
     <form>
       <h1>Customer Form</h1>
 
       <Input
         name="fullname"
-        label="full name"
+        label="Full name"
         value={props.student.fullname}
-        onChange={updateStudentFromUI.bind(this)}
+        onChange={updateStudentFromUI}
         error={(props.errors.fullname) ? props.errors.fullname.errorMessage : ""}
       />
 
       <Input
         name="email"
-        label="email"
+        label="Email"
         value={props.student.email}
-        onChange={updateStudentFromUI.bind(this)}
+        onChange={updateStudentFromUI}
         error={(props.errors.email) ? props.errors.email.errorMessage : ""}
       />
 
+      <Select
+        name="country"
+        label="Country"
+        value={props.student.country.id}
+        onChange={(e: any) => onSelectCountry(e, {id: e.target.value})}
+        error={(props.errors.country) ? props.errors.country.errorMessage : ""}
+        option={
```

Comment:
This should be called `options`.
Furthermore it could be replaced by an object-as-dictionary: `{ "id": "name", ... }` thus you get a cleaner usage pattern (like you don't need to repeat `key` and `value` everywhere) and more flexibility to change implementation eventually.